### ISSUE TITLE
Define app method for Rack::Test in rambo_helper.rb

### ISSUE THREAD

**Issue by danascheider** [danascheider/rambo#83]
## Summary

The `rambo_helper.rb` file is currently not adequate for non-Rails apps, both because it doesn't define an `app` method and because it doesn't include Rack::Test::Methods in the RSpec config. These things need to change, but before they can, we need to find a reliable way of getting the name of the application to set the `app` method output. With Rails applications, `Rails.application` does the job, but there is no analogous method for Sinatra or Grape.

## Expected Behavior

The Rambo helper should have code like the following (which is for a Grape app):
```ruby
module ApiHelper
  include Rack::Test::Methods

  def app
    PersonalCalculator::API
  end
end
```

## Current Behavior

Currently, there is no way for us to get the name of the application class (in this case, `PersonalCalculator::API`) into the `app` method definition in the `rambo_helper.rb` file, so this method is simply not defined.

## Possible Solution

Need to check the binding environment to see what classes are defined and what they inherit from. (In the case of Rails apps, `Rails.application` does this for us, and so does `Sinatra::Application` in classic Sinatra apps.) At this point we're looking at fairly substantial API changes, as follows.

Instead of having a `--no-rails` flag/`rails: false` option, I'm going to add a `--framework` option with the default value of `rails` (other options, at the moment, being Sinatra and Grape). This will facilitate use of Rambo for these 3 frameworks, unfortunately making it incompatible with others in the process. I encourage PRs extending support to other frameworks as needed.

In Rambo 0.6, the `app` method will be defined by identifying subclasses of the base class for each framework (i.e., `Sinatra::Application`, `Sinatra::Base`, and `Grape::API`). For this release, only one such subclass will be used. This means Rambo cannot currently be used for certain modular Sinatra apps or Grape apps with more than one `Grape::API` subclass. I will work on adding support for these use cases over time.

## Your Environment

* Version used: 0.5.0
* Sinatra: 1.4.7
* Grape: 0.16.2
* Grape app: https://github.com/danascheider/personal_calculator_grape
* Modular Sinatra app: https://github.com/danascheider/personal_calculator_sinatra_modular
* Classic Sinatra app: https://github.com/danascheider/personal_calculator_sinatra

**Comment by danascheider** [comment]
Cf #79

**Comment by danascheider** [comment]
Turns out, there is actually a `subclasses` method for all `Class` instances. This dramatically simplifies my life.

**Comment by danascheider** [comment]
Resolved in #84
